src/hooks/useExtraAPY.ts

```diff
@@ -1,20 +1,107 @@
 import Decimal from "decimal.js";
 
-import { getTotalSupplyAndBorrowUSD } from "../redux/assetsSelectors";
+import { getAccountPortfolio } from "../redux/accountSelectors";
+import { getAssets, getTotalSupplyAndBorrowUSD } from "../redux/assetsSelectors";
+import { computeDailyAmount } from "../redux/selectors/getAccountRewards";
+import { getStaking } from "../redux/selectors/getStaking";
+import { getConfig } from "../redux/appSelectors";
 import { useAppSelector } from "../redux/hooks";
+import { shrinkToken } from "../store/helper";
 
-export function useExtraAPY({ tokenId, isBorrow }) {
-  const [totalSupplyUSD, totalBorrowUSD] = useAppSelector(getTotalSupplyAndBorrowUSD(tokenId));
+export function useExtraAPY({ tokenId: assetId, isBorrow }) {
+  const [totalSupplyUSD, totalBorrowUSD] = useAppSelector(getTotalSupplyAndBorrowUSD(assetId));
+  const { xBRRR, extraXBRRRAmount } = useAppSelector(getStaking);
+  const portfolio = useAppSelector(getAccountPortfolio);
+  const appConfig = useAppSelector(getConfig);
+  const assets = useAppSelector(getAssets);
+  const asset = assets.data[assetId];
+  const assetDecimals = asset.metadata.decimals + asset.config.extra_decimals;
 
-  const computeRewardAPY = (rewardsPerDay, decimals, price) => {
-    return new Decimal(rewardsPerDay)
+  const computeRewardAPY = (rewardTokenId, rewardsPerDay, decimals, price) => {
+    const rewardAsset = assets.data[rewardTokenId];
+    const type = isBorrow ? "borrowed" : "supplied";
+    const farmData = portfolio.farms?.[type]?.[assetId]?.[rewardTokenId];
+
+    const totalDailyRewards = new Decimal(rewardsPerDay)
       .div(new Decimal(10).pow(decimals))
-      .mul(365)
-      .mul(price)
-      .div(isBorrow ? totalBorrowUSD : totalSupplyUSD)
-      .mul(100)
       .toNumber();
+
+    const totalDeposits = isBorrow ? totalBorrowUSD : totalSupplyUSD;
+
+    if (!farmData)
+      return new Decimal(rewardsPerDay)
+        .div(new Decimal(10).pow(decimals))
+        .mul(365)
+        .mul(price)
+        .div(totalDeposits)
+        .mul(100)
+        .toNumber();
+
+    const { multiplier, totalBoostedShares, shares } = computeDailyAmount(
+      type,
+      asset,
+      rewardAsset,
+      portfolio,
+      xBRRR + extraXBRRRAmount,
+      farmData,
+      appConfig.booster_decimals,
+    );
+
+    console.info(
+      isBorrow,
+      assetId,
+      rewardTokenId,
+      totalDailyRewards,
+      price,
+      multiplier,
+      totalBoostedShares,
+      shares,
+    );
+
+    // APY = total_daily_rewards * 365 * users_multiplier / (total_deposits * total_boosted_shares / total_shares)
+    const apy =
+      ((totalDailyRewards * price * 365 * multiplier) /
+        ((totalDeposits * totalBoostedShares) / shares)) *
+      100;
+
```

Comment:
Need to ask @evgenykuzyakov what shares is from the contract? Is it the "units" deposited, or has it already applied the current multiplier? 

If "shares" is units, then:

apy = ((totalDailyRewards * price * 365) * (shares * multiplier / totalBoostedShares)) / totalDeposits

if "shares" has applied current multiplier, then we need to first find the deposited units first, then apply the new multiplier to calculate the new boosted shares.